File: src/main/java/com/kyiv_kug/boxing/Boxing.java
package com.kyiv_kug.boxing;

public class Boxing {

    public static void main(String[] args) {
        print(0); // 0 int
        Integer arg = 1;
        print(arg); // 1 Integer
        print(new Integer(2)); // 2 Integer
        Integer argNull = null;
        print(argNull); // null Integer
    }

    private static void print(int a){
        System.out.println(a + " int");
    }

    private static void print(Integer a){
        System.out.println(a + " Integer");
    }
}

File: src/main/java/com/kyiv_kug/boxing/Boxing.kt
package com.kyiv_kug.boxing

fun main(args: Array<String>) {
    print(0)
    val arg = 1
    print(arg)
    print(2)
    val argNull: Int? = null
    print(argNull)
}

private fun print(a: Int) {
    println(a.toString() + " int")
}

private fun print(a: Int?) {
    println(a?.toString() + " Integer")
}



